kyuubi-ha/src/main/scala/org/apache/kyuubi/ha/client/zookeeper/ZookeeperClientProvider.scala

```diff
@@ -108,7 +108,10 @@ object ZookeeperClientProvider extends Logging {
           throw new IOException(s"${HA_ZK_AUTH_KEYTAB.key}: $keytab does not exists")
         }
         System.setProperty("zookeeper.sasl.clientconfig", "KyuubiZooKeeperClient")
-        val serverPrincipal = KyuubiHadoopUtils.getServerPrincipal(principal)
+        conf.get(HA_ZK_AUTH_SERVER_PRINCIPAL).foreach { zkServerPrincipal =>
+          System.setProperty("zookeeper.server.principal", zkServerPrincipal)
```

Comment:
ZOOKEEPER-1467: Make server principal configurable at client side.

https://issues.apache.org/jira/browse/ZOOKEEPER-1467

Fix Version/s: 3.6.0, 3.5.7

Reply:
Since #4852, it's allowed to use Zookeeper 3.6 client by enabling profile `-Pzookeeper-3.6` on the packaging, let me enrich the configuration docs to mention the requirements.